gpytorch/lazy/sum_lazy_variable.py

```diff
@@ -73,7 +72,7 @@ def __add__(self, other):
 
     def __getitem__(self, i):
         sliced_lazy_vars = [lazy_var.__getitem__(i) for lazy_var in self.lazy_vars]
-        return SumLazyVariable(*sliced_lazy_vars)
+        return sum(*sliced_lazy_vars)
```

Comment:
I don't know precisely what sum does internally, but is there a chance that using `sum(lazy_var_list)` is riskier than using `SumLazyVariable(lazy_var_list)`? I notice in __init__ you explicitly fix the case where someone is mixing and matching `Variables` and `LazyVariables` -- will sum do something silly where if a Variable appears before a `LazyVariable` in the list it will do `the_variable + the_lazy_variable`, which could fail because the add method for `Variables` does not expect `LazyVariables`?

Reply:
This is intentional. `__getitem__` may return LazyVariables, or it may return Variables. If the returned items are LazyVariables, then we'll get a sum lazy variable. Otherwise we'll get a LazyVariable.